pkg/cgroup/cgroup_stats.go

```diff
@@ -17,84 +17,110 @@ limitations under the License.
 package cgroup
 
 import (
-	"bufio"
 	"fmt"
-	"os"
-	"path/filepath"
-	"regexp"
-	"strconv"
 	"strings"
 
+	"github.com/containerd/cgroups"
+	statsv1 "github.com/containerd/cgroups/stats/v1"
+	"github.com/containerd/cgroups/v3/cgroup2"
+	statsv2 "github.com/containerd/cgroups/v3/cgroup2/stats"
+
 	"github.com/sustainable-computing-io/kepler/pkg/config"
 )
 
-const (
-	ioStatFile = "io.stat"
-	reIOStat   = `(\d+):(\d+).rbytes=(\d+).wbytes=(\d+)` // 8:16 rbytes=58032128 wbytes=0 rios=120 wios=0 dbytes=0 dios=0
-)
+func getcGroupStat(pid int) (interface{}, error) {
+	p := fmt.Sprintf(procPath, pid)
+	_, path, err := cgroups.ParseCgroupFileUnified(p)
+	if err != nil {
+		return nil, err
+	}
 
-var (
-	reIO = regexp.MustCompile(reIOStat)
-)
+	if config.GetCGroupVersion() == 1 {
+		control, err := cgroups.Load(cgroups.V1, cgroups.StaticPath(path))
+		if err != nil {
+			return nil, err
+		}
+		return control.Stat(cgroups.IgnoreNotExist)
+	} else {
+		str := strings.Split(path, "/")
+		size := len(str)
+		slice := strings.Join(str[0:size-1], "/") + "/"
+		group := str[size-1]
 
-func ReadAllCgroupIOStat() (rBytes, wBytes uint64, disks int, err error) {
-	path := filepath.Join(cgroupPath, ioStatFile)
-	return readIOStat(path)
+		control, err := cgroup2.LoadSystemd(slice, group)
+		if err != nil {
+			return nil, err
+		}
+		return control.Stat()
+	}
 }
 
-func ReadCgroupIOStat(cGroupID, pid uint64) (rBytes, wBytes uint64, disks int, err error) {
-	var path string
-	if config.EnabledEBPFCgroupID {
-		path, err = getPathFromcGroupID(cGroupID)
-	} else {
-		path, err = getPathFromPID(procPath, pid)
+func GetAvailableCgroupMetrics() []string {
```

Comment:
const? instead of function?

Reply:
We can return the content in a function, similar to the other functions employed [here](https://github.com/sustainable-computing-io/kepler/blob/main/pkg/collector/metric/stats.go#L41). Typically, it's suggested to employ Get and Set to access values from other packages. However, I agree that  we don't need to create a new variable for each function call, even if it's called just once. I will update this.